app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -87,4 +83,68 @@ public void onDisplayNetworkError(boolean networkNotAvailable) {
     public void onRequestFinish(@NotNull UiFinishReason uiFinishReason) {
         finish();
     }
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the ok/close button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorClose() {
+        onRequestFinish(UiFinishReason.UNABLE_TO_LOAD_SPA);
+    }
+
+    // -----WebUiNetworkErrorFragment.Listener Overrides -----
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the retry button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorRetry() {
+        loadWebUi();
+    }
+
+    private void loadWebUi() {
+        if (this.isFinishing()) {
```

Comment:
why the `this.`?
Also (up to you if you want to change) but we should be able to oneline this: `if (isFinishing) return;`

Reply:
adjusted